jerry-core/ecma/operations/ecma-exceptions.c

```diff
@@ -187,6 +198,113 @@ ecma_new_standard_error (ecma_standard_error_t error_type, /**< native error typ
   return new_error_obj_p;
 } /* ecma_new_standard_error */
 
+#if JERRY_BUILTIN_AGGREGATE_ERROR
+/**
+ * aggregate-error object constructor.
+ *
+ * @return newly constructed aggregate errors
+ */
+ecma_value_t
+ecma_new_aggregate_error (ecma_value_t error_list_val, /**< errors list */
+                          ecma_value_t message_val) /**< message string */
+{
+  ecma_object_t *new_error_object_p;
+
+  if (!ecma_is_value_undefined (message_val))
+  {
+    ecma_string_t *message_string_p = ecma_op_to_string (message_val);
+
+    if (JERRY_UNLIKELY (message_string_p == NULL))
+    {
+      return ECMA_VALUE_ERROR;
+    }
+
+    new_error_object_p = ecma_new_standard_error (ECMA_ERROR_AGGREGATE, message_string_p);
+    ecma_deref_ecma_string (message_string_p);
+  }
+  else
+  {
+    new_error_object_p = ecma_new_standard_error (ECMA_ERROR_AGGREGATE, NULL);
+  }
+
+  ecma_value_t using_iterator = ecma_op_get_method_by_symbol_id (error_list_val, LIT_GLOBAL_SYMBOL_ITERATOR);
+
+  if (ECMA_IS_VALUE_ERROR (using_iterator))
+  {
+    ecma_deref_object (new_error_object_p);
+    return using_iterator;
+  }
+
+  if (!ecma_is_value_undefined (using_iterator))
+  {
+    ecma_value_t next_method;
+    ecma_value_t iterator = ecma_op_get_iterator (error_list_val, using_iterator, &next_method);
+    ecma_free_value (using_iterator);
+
+    if (ECMA_IS_VALUE_ERROR (iterator))
+    {
+      ecma_deref_object (new_error_object_p);
+      return iterator;
+    }
+
+    ecma_collection_t *error_list = ecma_new_collection ();
+    ecma_value_t result = ECMA_VALUE_EMPTY;
+
+    while (true)
+    {
+      ecma_value_t next = ecma_op_iterator_step (iterator, next_method);
+
+      if (ECMA_IS_VALUE_ERROR (next))
+      {
+        result = next;
+        break;
+      }
+
+      if (next == ECMA_VALUE_FALSE)
+      {
+        break;
+      }
+
+      /* 8.e.iii */
+      ecma_value_t next_error = ecma_op_iterator_value (next);
+      ecma_free_value (next);
+
+      if (ECMA_IS_VALUE_ERROR (next_error))
+      {
+        result = next_error;
+        break;
+      }
+
+      ecma_collection_push_back (error_list, next_error);
+    }
+
+    ecma_free_value (iterator);
+    ecma_free_value (next_method);
+
+    if (ECMA_IS_VALUE_ERROR (result))
+    {
+      ecma_collection_free (error_list);
+      ecma_deref_object (new_error_object_p);
+      return result;
+    }
+
+    ecma_free_value (result);
+
+    ecma_value_t error_list_arr = ecma_op_new_array_object_from_collection (error_list, true);
+    ecma_property_value_t *prop_value_p;
+    prop_value_p = ecma_create_named_data_property (new_error_object_p,
+                                                    ecma_get_magic_string (LIT_MAGIC_STRING_ERRORS_UL),
+                                                    ECMA_PROPERTY_CONFIGURABLE_WRITABLE,
+                                                    NULL);
+    prop_value_p->value = ecma_copy_value_if_not_object (error_list_arr);
```

Comment:
error_list_arr must be an object, so simply `prop_value_p->value = error_list_arr`